plugins/npm/src/tasks/npm-publish.ts

```diff
@@ -0,0 +1,82 @@
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { Task } from '@dotcom-tool-kit/types'
+import pacote from 'pacote'
+import { readState } from '@dotcom-tool-kit/state'
+import pack from 'libnpmpack'
+import { publish } from 'libnpmpublish'
+import { styles } from '@dotcom-tool-kit/logger'
+import tar from 'tar'
+import { PassThrough as PassThroughStream } from 'stream';
+
+export const semVerRegex = /^v\d+\.\d+\.\d+(-.+)?/
+export const prereleaseRegex = /^v\d+\.\d+\.\d+(?:-\w+\.\d+)$/
+export const releaseRegex = /^v\d+\.\d+\.\d+$/
+
+type TagType = "prerelease" | "latest"
+
+export default class NpmPublish extends Task {
+  static description = ''
+  
+  getNpmTag(tag: string): TagType {
+    if(!tag) {
+        throw new ToolKitError('CIRCLE_TAG environment variable not found. Make sure you are running this on a release version!')
+    }
+    if(prereleaseRegex.test(tag)) {
+      return 'prerelease'
+    }
+    if(releaseRegex.test(tag)) { 
+      return 'latest'
+    }
+    throw new ToolKitError(`CIRCLE_TAG does not match regex ${semVerRegex}. Configure your release version to match the regex eg. v1.2.3-beta.8`)
+  }
+
+  async listPackedFiles(tarball: Buffer): Promise<void> {
+    this.logger.info('packed files:')
+
+    new PassThroughStream()
+      .end(tarball)
+      .pipe(tar.t({onentry: entry => this.logger.info(`- ${styles.filepath(entry.header.path)}`)}))
+  }
+
+  async run(): Promise<void> {
+    this.logger.info('preparing to publish your npm package....')
+
+    const packagePath = process.cwd()
+    const manifest = await pacote.manifest(packagePath)
+
+    const ci = readState('ci')
+    
+    if(!ci) {
+      throw new ToolKitError(
+        `Could not find state for ci, check that ${styles.hook('publish:tag')} ran successfully`
+      )
+    }
+
+    const tag = ci.tag
+
+    const npmTag = this.getNpmTag(tag)
+    
+    this.logger.info(`version ${tag} ready to be published with ${npmTag} tag`)
+
+    if (!process.env.NPM_AUTH_TOKEN) {
+        throw new ToolKitError('NPM_AUTH_TOKEN environment variable not found! Make sure your project is pulling in the env vars from /teams/next/circleci/component')
```

Comment:
**suggestion**: this should probably be updated to suggest checking the new GitHub contexts rather than Vault now